test/unit/specs/hive/map.js

```diff
@@ -673,36 +684,43 @@ describe('Hive - Map Store', function() {
       })
 
       it('fails with invalid value', async () => {
-        const errorMsg = 'Value must be provided and must be a string.'
+        const errorMsg = 'Value must be provided and must be one of types: string, number, boolean, object, array.'
 
         await expect(() => store.set('k', undefined)).to.throw(errorMsg)
         await expect(() => store.set('k', null)).to.throw(errorMsg)
-        await expect(() => store.set('k', false)).to.throw(errorMsg)
-        await expect(() => store.set('k', 0)).to.throw(errorMsg)
-        await expect(() => store.set('k', '')).to.throw(errorMsg)
-        await expect(() => store.set('k', true)).to.throw(errorMsg)
-        await expect(() => store.set('k', 123)).to.throw(errorMsg)
-        await expect(() => store.set('k', {})).to.throw(errorMsg)
-        await expect(() => store.set('k', [])).to.throw(errorMsg)
-        await expect(() => store.set('k', () => undefined)).to.throw(errorMsg)
+        await expect(() => store.set('k', () => true)).to.throw(errorMsg)
+        await expect(() => store.set('k', 10n)).to.throw(errorMsg)
+        await expect(() => store.set('k', Symbol('id'))).to.throw(errorMsg)
+        await expect(() => store.set('k', [10n])).to.throw(errorMsg)
       })
     })
 
     describe('Set With Overwrite', async () => {
-      it('success', async () => {
-        const request = prepareMockRequest(fakeResult)
+      it('success values', async () => {
+        async function testValidValue(value) {
+          const request = prepareMockRequest(fakeResult)
 
-        const result = await store.setWithOverwrite('target', 'value1')
+          const result = await store.setWithOverwrite('target', value)
 
-        expect(request).to.deep.include({
-          method: 'PUT',
-          path  : `${APP_PATH}/hive/${hiveName}/map/${storeKey}/set-with-overwrite/target`,
-          body  : {
-            value: 'value1'
-          }
-        })
+          expect(request).to.deep.include({
+            method: 'PUT',
+            path  : `${APP_PATH}/hive/${hiveName}/map/${storeKey}/set-with-overwrite/target`,
+            body  : {
+              value: value
+            }
+          })
 
-        expect(result).to.be.eql(fakeResult)
+          expect(result).to.be.eql(fakeResult)
+        }
+
+        await testValidValue('string')
```

Comment:
wrapping `expect` into a function or loop is a bad practice, when it fails it will be tough to understand what/where it exactly fails

Reply:
I tried to avoide 100+ "it". Or do it with "it"?

Reply:
do it in a single it